cmd/tailscale/cli/up.go

```diff
@@ -498,7 +498,23 @@ func runUp(ctx context.Context, cmd string, args []string, upArgs upArgsT) (retE
 
 	var printed bool // whether we've yet printed anything to stdout or stderr
 	var loginOnce sync.Once
-	startLoginInteractive := func() { loginOnce.Do(func() { localClient.StartLoginInteractive(ctx) }) }
+
+	// localAPIMu should be held while doing mutable LocalAPI calls
+	// to the backend. In particular, it prevents StartLoginInteractive from
+	// being called from the watcher goroutine while the Start call from
+	// the other goroutine is in progress.
+	// See https://github.com/tailscale/tailscale/issues/7036#issuecomment-2053771466
+	// TODO(bradfitz): simplify this once #11649 is cleaned up and Start is
+	// hopefully removed.
+	var localAPIMu sync.Mutex
+
+	startLoginInteractive := func() {
+		loginOnce.Do(func() {
```

Comment:
you could probably change this to a `sync.OnceFunc`

Reply:
oh yeah, nice, done